### PR TITLE
frontmatter, relative images, other meta

### PR DESCRIPTION
**Pull request by wuntusk**

### CODE REVIEW

**Comment by stanislawosinski** [comment]
Hi @wuntusk, thanks for the PR! Can you split the commit into the commit that introduces the actual changes and another one that changes the formatting? (Or skip the latter altogether.) It's hard to follow the diff when most of the commit contains formatting changes.

**Comment by wuntusk** [comment]
@stanislawosinski Sorry I submitted this by accident.. thus the reason I closed it out.  These changes were for my own use case some of which would probably break yours.  I'm using this to convert an old web site with about 700 individual folders each which holds an html page and it's assets in that folder. What I was doing here was 
1. Allow images to live in a relative location to the html.
2. Create frontmatter and add the following
- title from the html title
- meta for every meta tag .. in my case I have 
```
    <meta name="author" content="Tigole"></meta>
    <meta name="date" content="2000-10-28" scheme="YYYY-MM-DD"></meta
```

   This allows me to generate an index page

3. I removed the extra formatting that gets added to each image as I was trying to retain the look of the original pages.
4. accidentally left some console logs in there
5. replace-variables crashes dereffing a null in the CreateMapReplacer which I was trying to track down via the check for null map and the console log which I haven't gotten back to 

If you're interested in the changes I can go try and fix this up.  The formatting probably got messed up as I'm using this in the plugins folder as a submodule and was editing from the project above which is using different lint/prettier rules.

**Comment by stanislawosinski** [comment]
Oh, that makes sense now! It looks like a very specific case, so let's leave that out of the main code.
